src/mol-model/structure/structure/mapping.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2024 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Alexander Rose <alexander.rose@weirdbyte.de>
+ */
+
+import { OrderedSet } from '../../../mol-data/int/ordered-set';
+import { ElementIndex } from '../model/indexing';
+import { Structure } from './structure';
+import { Unit } from './unit';
+
+/** Serial index of an element in the structure across all units */
+export type SerialIndex = { readonly '@type': 'serial-index' } & number
+
+export interface SerialMapping {
+    /** Cumulative count of preceding elements for each unit */
+    cumulativeUnitElementCount: ArrayLike<number>
+    /** Unit index for each serial element in the structure */
+    unitIndices: ArrayLike<number>
+    /** Element index for each serial element in the structure */
+    elementIndices: ArrayLike<ElementIndex>
+    /** Get serial index of element in the structure */
+    getSerialIndex: (unit: Unit, element: ElementIndex) => SerialIndex
+}
+
+export function getSerialMapping(structure: Structure): SerialMapping {
+    const { units, elementCount, unitIndexMap } = structure;
+    const cumulativeUnitElementCount = new Uint32Array(units.length);
+    const unitIndices = new Uint32Array(elementCount);
```

Comment:
```suggestion
    const unitIndices = new Uint32Array(elementCount);
```

If unit count is <65536 it can be `Uint16`